### DISCUSSION TITLE
Beginner-friendly User Guide

### DISCUSSION THREAD

**Discussion by aleewen** [PyO3/pyo3#4071] [Show and tell]
Hi all,

I found the official PyO3 user guide to be quite heavy and difficult to read as a beginner to the library; documentation for the simple features and functionality were mixed up with the complicated ones. As my method of note-taking, I wrote a user guide that I find to be a little more beginner friendly that is much easier to get started with.

[Rust_ PyO3.md](https://github.com/aleewen/guides/blob/main/Rust_%20PyO3.md)

**Comment by adamreichold** [comment]
> documentation for the simple features and functionality were mixed up with the complicated ones. 

FWIW, I agree with this. As is often the case, documentation is somewhat unloved and especially curating the existing snippets of information into more coherent stories targeted at different backgrounds is a large amount of work.

Maybe you could propose an even further slimmed down version of your notes as a tutorial or quick start section of our main guide?

**Reply by aleewen** [reply]
Sure thing. How should I go about proposing that? I think a Quick Start guide would be very nice to those who are just getting started, and if they get advanced enough to want more than what is there, they can also refer to the full user guide.

**Reply by adamreichold** [reply]
Personally, I would just add this as a new chapter to the existing guide. This way, moving on to the more complex stuff is easy. Also, we do not need to worry about how to name/where to host/etc.

**Reply by MusicalNinjaDad** [reply]
I felt the same and had made a mental note to help with putting something together to make it easier to get started... I had mainly noted topics around:
1. setting up the dev environment (I have a devcontainer I'd happily share to make that easier)
2. structuring the project (rust code, wrapped code, python code, rust & python tests)
3. testing (what is easiest to test where & how) [I really love the unloved - documentation about testing!]

I'd be happy to contribute these into a relevant section of the documentation as a service to the community.

@aleewen  - that guide of yours looks nice and simple to follow.
